app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java

```diff
@@ -245,93 +247,111 @@ protected void setBarVisible(ViewGroup barLayout, boolean visible) {
         }
     }
 
-    public static class TextSelection {
+    protected void runRegularPrefixAction(String action) {
+        runRegularPrefixAction(action, null, false);
+    }
+
+    protected void runRegularPrefixAction(String action, Boolean ignoreIndent) {
+        runRegularPrefixAction(action, null, ignoreIndent);
+    }
 
-        private int _selectionStart;
-        private int _selectionEnd;
-        private Editable _editable;
+    protected void runRegularPrefixAction(String action, String replaceString) {
+        runRegularPrefixAction(action, replaceString, false);
+    }
 
+    protected void runRegularPrefixAction(final String action, final String replaceString, final Boolean ignoreIndent) {
 
-        TextSelection(int start, int end, Editable editable) {
-            _selectionStart = start;
-            _selectionEnd = end;
-            _editable = editable;
-        }
+        String replacement = (replaceString == null)? "" : replaceString;
 
-        private void insertText(int location, String text) {
-            _editable.insert(location, text);
-            _selectionEnd += text.length();
-        }
+        String patternIndent = ignoreIndent ? "(^\\s*)" : "(^)";
+        String replaceIndent = "$1";
 
-        private void removeText(int location, String text) {
-            _editable.delete(location, location + text.length());
-            _selectionEnd -= text.length();
+        String escapedAction = String.format("\\Q%s\\E", action);
+        String escapedReplace = String.format("(\\Q%s\\E)?", replacement);
+
```

Comment:
Rewrote prefix action to work in terms of regex action.

In this future, this can be used to change between list types etc
i.e.`- foo` -> `- [ ] foo` rather than `- foo` -> `- [ ] - foo`